### PR TITLE
EnterpriseContractPolicy CRD

### PR DESCRIPTION
**Pull request by zregvart**
Cretes the EnterpriseContractPolicy CRD from the enterprise-contract-controller[1] repository.

Ref. https://issues.redhat.com/browse/HACBS-244

[1] https://github.com/hacbs-contract/enterprise-contract-controller

### CODE REVIEW

File: argo-cd-apps/base/enterprise-contract.yaml [hunk 1]
```diff
@@ -44,3 +44,32 @@ spec:
...
+  name: enterprise-contract
+
+spec:
+  project: default
+
+  source:
+    path: components/enterprise-contract
+    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
+
+  destination:
```
[the remark this answers is not in the document]
Comment (zregvart) [reply argo-cd-apps/base/enterprise-contract.yaml:60]: Not for a CRD, it's cluster-wide (non-namespaced).

**Review by joejstuart — APPROVED** [review]

File: components/enterprise-contract/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- https://github.com/hacbs-contract/enterprise-contract-controller/config/crd
```
Comment (Michkov) [inline components/enterprise-contract/kustomization.yaml:5]: please point directly to commit, we do not want to have untrackted updates.

**Pull request merged**